pandas/core/groupby/generic.py

```diff
@@ -583,6 +583,8 @@ def _choose_path(self, fast_path, slow_path, group):
         try:
             res_fast = fast_path(group)
 
+            if res_fast.columns != group.columns:
```

Comment:
can you put after the comment, does if matter if we put this after the shape check?

and wont' ``getattr(res, 'columns', None) != getattr(res_fast, 'columns', None)`` work?

Reply:
I don't think it matters if it is after the shape change. I'll move it. Why do you want to use `getattr`? Why would columns not be set?

Reply:
Actually, it does. Checking columns is probably much faster than the contents of the shape check, so we should do it first to avoid paying that cost every time.

Reply:
is ``group`` *always* a DataFrame here?

Reply:
Yes. I tested this hypothesis by asserting that it is a dataframe in all tests.

Reply:
Also just confirmed that this code *does not* get called when the user writes `df.groupby('group').column_name.transform(func)`. So I am very confident that `group` is always a `pd.DataFrame` here.